packages/shared/src/utils/analytics/rudderstack.ts

```diff
@@ -0,0 +1,157 @@
+import * as rudderanalytics from 'rudder-sdk-js';
+import { isProduction } from '../config';
+
+type SignupProvider = 'email' | 'phone' | 'google' | 'facebook' | 'apple';
+
+type VirtualSignupFormAction = {
+    action:
+        | 'open'
+        | 'started'
+        | 'email_confirmed'
+        | 'signup_continued'
+        | 'country_selection_screen_opened'
+        | 'password_screen_opened'
+        | 'signup_done'
+        | 'signup_flow_error'
+        | 'go_to_login';
+    signup_provider?: SignupProvider;
+    form_source?: string;
+    form_name?: string;
+    error_message?: string;
+};
+
+type RealAccountSignupFormAction = {
+    action:
+        | 'open'
+        | 'step_passed'
+        | 'save'
+        | 'restore'
+        | 'close'
+        | 'real_signup_error'
+        | 'other_error'
+        | 'real_signup_finished';
+    step_codename?: string;
+    step_num?: number;
+    user_choice?: string;
+    source?: string;
+    form_name?: string;
+    real_signup_error_message?: string;
+    landing_company: string;
+};
+
+type VirtualSignupEmailConfirmationAction = {
+    action: 'received' | 'expired' | 'confirmed' | 'error';
+    signup_provider?: SignupProvider;
+    form_source?: string;
+    email_md5?: string;
+    error_message?: string;
+};
+
+type TradeTypesFormAction =
```

Comment:
@adrienne-deriv Let's do some type improvements together when you are free if you like 🙇🏻

Reply:
Yeaah I need some feedback on these types 😅